src/code/reporting/components/app.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import { ProcessCSVData } from "../process-sage";
+import { XSLtoCSV } from "../process-xlsx";
+
+export interface IDownloadLinkProps {
+  csvString: string;
+  fileName: string;
+}
+
+const DownloadLink = (props: IDownloadLinkProps) => {
+  const {csvString, fileName} = props;
+  const blob = new Blob([csvString], { type: "text/csv" });
+  const downloadURL = window.URL.createObjectURL(blob);
+  if (csvString && csvString.length > 0) {
+    return (
+      <div>
+        <a href={downloadURL} download={fileName}>
+          download {fileName}
+        </a>
+      </div>
+    );
+  }
+  return(null);
+};
+
+interface ICSVFileRecord {
+  content: string;
+  fileName: string;
+}
+
+export interface IAppProps {
+  userName: string;
+  place: string;
+}
+
+export interface IAppState {
+  CSVs: ICSVFileRecord[];
+  status: string;
+  running: boolean;
+}
+
+export class ReportApp extends React.Component<IAppProps, IAppState> {
+
+  public static displayName = "AboutView";
+
+  public state: IAppState = {
+    CSVs: [],
+    status: "starting",
+    running: false,
+  };
+
+  public render() {
+    const {running, status, CSVs} = this.state;
+    const onChangeFile = (e) => this.handleFileChange(e);
```

Comment:
You could also do `this.handleFileChange = this.handleFileChange.bind(this)` in a constructor.